hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/lock/OmLockInfo.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.lock;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.hadoop.hdds.utils.LockInfo;
+import org.apache.hadoop.ozone.OzoneConsts;
+
+/**
+ * This class represents the locking information that is required by an OM request.
+ * Requests can define whether they need volume, bucket, or key locks of read or write types.
+ * A request gatekeeper can handle these requirements to identify conflicting requests.
+ *
+ * Sample usage:
+ * {@link org.apache.hadoop.ozone.om.request.OMClientRequest} adds a required method {@code getLockInfo} which returns
+ * an instance of this class. All OM requests then implement this method and return an instance of this class to define
+ * their locking requirements.
+ *
+ * Note that this class defines the interface for each of the ~100 OM requests to communicate their requirements to the
+ * gatekeeper. Any breaking changes to this interface will have a large ripple effect.
+ */
+public final class OmLockInfo {
+  private final LockInfo volumeLock;
+  private final LockInfo bucketLock;
+  private final Set<LockInfo> keyLocks;
```

Comment:
If we require locking on two or more volumes/buckets later on, it will require a lot of code change.